File: function/hello.py
import time
import random

def main(args):
    name = args.get("time", 0)
    if name != 0:
        time.sleep(float(name) / 1000)
    else:
        probability = random.random()
        if probability <= 0.2:
            sleep_time = random.uniform(0.001, 0.1)  # 1ms到100ms
        elif probability <= 0.8:
            sleep_time = random.uniform(0.1, 1)  # 100ms到1s
        else:
            sleep_time = random.uniform(1, 3)  # 1s到3s
        time.sleep(sleep_time)
    return {"greeting": name}

File: function/hello.js
function sleep(ms) {
    return new Promise(resolve=>setTimeout(resolve, ms))
}

async function sleepRandomDuration() {
    const random = Math.floor(Math.random() * 100); // 生成0到99的随机数

    if (random < 20) { // 20%的概率
        const sleepDuration = Math.floor(Math.random() * 100) + 1; // 1ms到100ms的随机数
        await sleep(sleepDuration);
    } else if (random < 80) { // 60%的概率
        const sleepDuration = Math.floor(Math.random() * 901) + 100; // 100ms到1s的随机数
        await sleep(sleepDuration);
    } else { // 20%的概率
        const sleepDuration = Math.floor(Math.random() * 2001) + 1000; // 1s到3s的随机数
        await sleep(sleepDuration);
    }
}

/**
 * @params is a JSON object with optional fields "name" and "place".
 * @return a JSON object containing the message in a field called "msg".
 */
async function main(params) {
  
    // if a value for name is provided, use it else use a default

    if (params.time !== undefined && params.time != null) {
        await sleep(params.time);
    } else {
        await sleepRandomDuration();
    }
    // construct the message using the values for name and place
    return {msg:  'Hello, World'};
}